### PR TITLE
bots: Move openshift image to CentOS-7

### PR DESCRIPTION
**Pull request by martinpitt**
Fedora 28 is EOL. As we still need Python 2 here, use CentOS 7 as a base
image (which will be around for a while).

Drop obsolete GnuTLS/Go certificate hack, current 3.11 OpenShift
versions have this fixed properly.

Move to virt-install-fedora to get rid of the libguestfs dependency and
be able to build this image in an unprivileged podman container.

 * [x] image-refresh openshift

### CODE REVIEW

**Comment by martinpitt** [comment]
I built this locally and successfully ran check-openshift in the rhel-7.8 branch against it.

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-02.
Log: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-110909/

**Comment by cockpituous** [comment]
Task failed: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-110909/

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-szjd1.
Log: https://logs-https-cockpit.apps.ci.centos.org/logs/image-refresh-12693-20190903-121818/

**Comment by cockpituous** [comment]
image-refresh openshift done: https://github.com/cockpituous/cockpit/commits/image-refresh-openshift-20190903-123636

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-02.
Log: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-132121/

**Comment by cockpituous** [comment]
Task failed: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-132121/

**Comment by cockpituous** [comment]
image-refresh in progress on 3-ci-srv-04.
Log: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-134242/

**Comment by cockpituous** [comment]
Task failed: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-134242/

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-04.
Log: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-145252/

**Comment by cockpituous** [comment]
Task failed: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-145252/

**Comment by martinpitt** [comment]
This `BlockingIOError: [Errno 11] Resource temporarily unavailable` is unnerving.. I pushed a fix to avoid this giant json log spew, which ought to help.

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-05.
Log: https://209.132.184.41:8493/logs/image-refresh-12693-20190903-154444/

**Comment by cockpituous** [comment]
image-refresh openshift done: https://github.com/cockpituous/cockpit/commits/image-refresh-openshift-20190903-163535

**Comment by martinpitt** [comment]
The "busy NFS" failure is just a flake, it works fine locally and also doesn't happen every time on CI. The [testNodeNavigation](https://209.132.184.41:8493/logs/pull-12693-20190903-195858-22802579-cockpit-project-cockpit-rhel-7.8-rhel-7-8/log.html#51) failure is real, though. I fixed it in PR #12700. So  we somehow need to land this PR first, and then re-run and land 12700 to get things back in order.

FTR, I tried to update openshift to Fedora 30, but it's too broken. OpenShift 3.x really wants to live in the RHEL world, so CentOS-7 is a much better base for it. :-)

File: bots/images/scripts/openshift.setup [hunk 1]
```diff
@@ -42,17 +42,14 @@ printf "OPENSHIFT LISTENING ON LOCALHOST\n  $ ssh -NL 8443:localhost:8443 root@1
 ln -sf ../selinux/config /etc/sysconfig/selinux
 printf 'SELINUX=permissive\nSELINUXTYPE=targeted\n' > /etc/selinux/config
 setenforce 0
-systemctl stop firewalld
-dnf mark install iptables
-dnf -y remove firewalld
```
Comment (marusak) [inline bots/images/scripts/openshift.setup]: any reason for dropping these two lines?

Comment (martinpitt) [reply bots/images/scripts/openshift.setup]: I just stumbled over them when I was going over the `dnf` commands. Disabling firewalld is faster and more robust, and it then will still exist for possible experimentation.

**Pull request merged**